### DISCUSSION TITLE
Unable to set SplitButton icon color

### DISCUSSION THREAD

**Discussion by DarkMortainius** [AvaloniaUI/Avalonia#15343] [Q&A]
Am I missing something, or is there no way to set the icon color in a SplitButton? The PathIcon used in the simple and fluent themes don't seem to use the foreground color at all.

**Comment by robloo** [comment]
The `Foreground` is usually inherited and is set in the Fluent theme. The issue is in how `PathIcon` is configured to override that color and set it directly (see below). That itself is an issue probably inherited from upstream WinUI.

https://github.com/AvaloniaUI/Avalonia/blob/ca520a233264208ccb489cd16d3ed2d1bda38883/src/Avalonia.Themes.Fluent/Controls/PathIcon.xaml#L13

Which definitely prevents the Fluent `SplitButton` theme from working correctly for pointer over on the `PathIcon` as can be seen below:

![capturegif](https://github.com/AvaloniaUI/Avalonia/assets/17993847/730eb5a3-aaf5-4859-8530-bc56e7593951)

I think we have have removed this hard-coded `Foreground` from most other places by now (except Text controls) and it should be removed from `PathIcon` as well. Not sure any potential effects for breaking changes.

Fixed, everything works as expected.
![capturegif](https://github.com/AvaloniaUI/Avalonia/assets/17993847/4eaaf05d-9498-4858-96a2-f7742ee58e4e)

You can then properly change the foreground colors of the split button using the lightweight styling resources. Keep in mind the foreground is shared for both the primary and secondary parts of the button though. So you can't just change the PathIcon by itself unless you re-template.

https://github.com/AvaloniaUI/Avalonia/blob/ca520a233264208ccb489cd16d3ed2d1bda38883/src/Avalonia.Themes.Fluent/Accents/FluentControlResources.xaml#L746-L756

**Reply by robloo** [reply]
@maxkatz6 How should I fix this?

 1. Are you OK with removing the hard-coded Foreground in PathIcon? (allowing it to be inherited)
 2. If not for concern of breaking changes, I can set the PathIcon Foreground in the SplitButton control theme directly.

num.1 seems like the best general-purpose solution.
